docs/spec/Interfaces.adoc

```diff
@@ -0,0 +1,91 @@
+== Interfaces
```

Comment:
Can we rename this section to "Port Interfaces" and consistently use the term "port interface"? I think it's OK to use "interface" as a shorthand for "port interface" from time to time, but I think we shouldn't do that too often.